### PR TITLE
LEP-438 partner integration

### PR DESCRIPTION
**Pull request by starswan**
## What does this pull request do?

Add hooks for partner data

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

**Review by ripan — APPROVED** [review]

File: cla_backend/libs/eligibility_calculator/calculator.py [hunk 1]
```diff
@@ -365,7 +365,8 @@ def filter_to_keys(dict_, keys):
                 logger.error("CFE and legacy_check() calcs disagree!\nCFE:    %s\nLegacy: %s" % (filter_out_zeros(cfe_calcs), filter_out_zeros(self.calcs)))
 
         # Gradual cut-over from using legacy_result to cfe_result
-        if self._is_non_means_tested(self.case_data) or self._without_partner(self.case_data):
+        if (self._is_non_means_tested(self.case_data)
+                or self._without_partner(self.case_data) or self._with_partner(self.case_data)):
```
Comment (davidread) [inline cla_backend/libs/eligibility_calculator/calculator.py:369]: once we merge LEP-441, we can get rid of the if completely 👍

File: cla_backend/libs/eligibility_calculator/calculator.py [hunk 2]
```diff
@@ -386,16 +384,27 @@ def _pounds_to_pence(value):
     def _without_partner(case_data):
         return hasattr(case_data.facts, "has_partner") and not case_data.facts.has_partner
 
+    @staticmethod
+    def _with_partner(case_data):
```
Comment (davidread) [inline cla_backend/libs/eligibility_calculator/calculator.py:388]: method no longer used

Comment (davidread) [reply cla_backend/libs/eligibility_calculator/calculator.py:388]: Also _without_partner

File: cla_backend/libs/eligibility_calculator/calculator.py [hunk 3]
```diff
@@ -421,33 +429,33 @@ def _is_data_complete_enough_to_call_cfe(case_data):
...
 
     @staticmethod
     def _translate_case(case_data, submission_date=None):
-        '''Translates CLA's CaseData to CFE-Civil request JSON'''
+        """Translates CLA's CaseData to CFE-Civil request JSON"""
         if not submission_date:
             submission_date = datetime.date.today()
-        # produce the simplest possible plain request to CFE to prove the route
+        # default DOB to 'nothing special' 40 years old rules-wise
+        default_dob = str(datetime.date(submission_date.year - 40, submission_date.month, submission_date.day))
```
Comment (davidread) [inline cla_backend/libs/eligibility_calculator/calculator.py:452]: Needs something like this to cope with leap years:
```suggestion
        default_dob = str(submission_date - relativedelta(years=40))
```

Comment (starswan) [reply cla_backend/libs/eligibility_calculator/calculator.py:444]: I think this line of code can go. Didn't realise it was still present

File: cla_backend/libs/eligibility_calculator/calculator.py [hunk 4]
```diff
@@ -404,6 +391,11 @@ def _is_non_means_tested(case_data):
         return case_data.facts.on_nass_benefits and case_data.category == "immigration"
 
     def _do_cfe_civil_check(self):
+        # This doesn't appear to be called from anywhere, but keep it for compatibility
```
Comment (davidread) [inline cla_backend/libs/eligibility_calculator/calculator.py:394]:
```suggestion
        # This property is not used by clients, but we support it while it's in the API.
```

**Review by davidread — APPROVED** [review]

**Pull request merged**